### ISSUE TITLE
Automatic reallocation failed

### ISSUE THREAD

**Issue by harperjf** [lfortran/lfortran#5739]
In this program the second assignment to ```x``` requires ```x``` to be reallocated because its shape has changed.
```
  integer,allocatable:: x(:)
  x = [1,2,3,4]
  print "(*(1X,I0))",x
  x = [5,6]
  print "(*(1X,I0))",x
end program
```
Lfortran 0.42.0 output:
```
Segmentation fault (core dumped)
```
Four other compilers gave this output, which IMHO is correct:
```
 1 2 3 4
 5 6
```

**Comment by harperjf** [comment]
As #5707 seemed to require the absence of a ```program``` statement I tried adding one to my little program, making it read
```
program test
  integer,allocatable:: x(:)
  x = [1,2,3,4]
  print "(*(1X,I0))",x
  x = [5,6]
  print "(*(1X,I0))",x
end program
```
Same output as before.

**Comment by kmr-srbh** [comment]
We do not support implicit allocation by default. You can do so using the `--realloc-lhs` option.

```console
(base) saurabh-kumar@Awadh:~/Projects/System/lfortran$ lfortran test.f90 --realloc-lhs
 1
 2
 3
 4

 5
 6
```

**Comment by kmr-srbh** [comment]
@harperjf yes, I think we have some ongoing work to fix printing.

**Comment by assem2002** [comment]
the printing issue should be fixed by https://github.com/lfortran/lfortran/pull/5039.

**Comment by harperjf** [comment]
Having been misled by assuming that the ```--std=f23``` option would enforce compliance with the Fortran standard I suggest that using that option should automatically use all the other options that make lfortran accept programs that comply with the standard, whether or not they agree with the opinions of lfortran's developers on what is good practice. I think the relevant options are
```
--realloc-lhs 
--use-loop-variable-after-loop 
--implicit-typing
--implicit-interface
```
but I may have missed some, or included some that would allow a non-conforming program.
If that is unacceptable to the developers I suggest that those options and ```--std=f23``` be included in a new option, possibly called ```--strict```

**Comment by certik** [comment]
Yes, that's a bug to not include all standard conforming options. Here are the current options: https://github.com/lfortran/lfortran/blob/main/src/bin/lfortran.cpp#L2336, the implicit ones are already enabled, so just the `--realloc-lhs` must be added, and also `--use-loop-variable-after-loop` (if it is conforming).
